examples/bri-3/src/bri/merkleTree/agents/merkleTree.agent.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import MerkleTree from 'merkletreejs';
+import { v4 } from 'uuid';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+import { MerkleTreeStorageAgent } from './merkleTreeStorage.agent';
+import * as crypto from 'crypto';
+
+@Injectable()
+export class MerkleTreeAgent {
+  constructor(private storageAgent: MerkleTreeStorageAgent) {}
+  public createNewMerkleTree(
+    leaves: string[],
+    hashFunction: string,
+  ): BpiMerkleTree {
+    const hashHelper = (data: any): Buffer => {
+      return crypto.createHash(hashFunction).update(data).digest();
+    };
+
+    const hashedLeaves = leaves.map((x) => hashHelper(x));
+
+    return new BpiMerkleTree(v4(), new MerkleTree(hashedLeaves, hashHelper));
+  }
+
+  public updateMerkleTree(
+    merkleTreeToUpdate: BpiMerkleTree,
+    leaves: string[],
+    hashFunction: string,
+  ) {
+    const hashHelper = (data: any): Buffer => {
+      return crypto.createHash(hashFunction).update(data).digest();
+    };
+
+    const hashedLeaves = leaves.map((x) => hashHelper(x));
+
+    merkleTreeToUpdate.updateMerkleTree(hashedLeaves, hashHelper);
+  }
+
+  public async fetchMerkleTreeCandidateByIdAndThrowIfValidationFails(
+    id: string,
+  ): Promise<BpiMerkleTree> {
+    const merkleTreeCandidate: BpiMerkleTree =
+      await this.storageAgent.getMerkleTreeById(id);
```

Comment:
Please refactor so we do not throw exception in the storage agent as its method might be invoked in a context where HTTP exceptions should not be thrown (i.e. background service such as nats messaging handler)

Reply:
Ah, now i see that we do not throw anywhere. We should throw here if merkleTreeCandidate is null